### PR TITLE
refactor: rename to pinyin, reorganize tests, fix fu bugs

### PR DESCRIPTION
**Pull request by mmdtoycar**
## Summary
- Rename all Japanese romaji identifiers to Chinese pinyin (shuntsu→shunzi, koutsu→kezi, etc.)
- Reorganize test files into `__tests__/` folders for both guobiao and riichi
- Fix 暗杠 fu calculation bug (was 8, should be 16 for 中张)
- Add 切上満貫 (kiriage mangan): 4番30符+ and 3番60符+ round up to mangan
- Fix wait type ambiguity: `generateTries` + `completedByWin` ensures best interpretation wins
- Refactor score.ts: extract `getBasePoints()`, remove dead `basePoints` field, named constants
- Add 7 new test cases for fu edge cases and wait ambiguity

## Test plan
- [ ] Run all 296 tests: `cd ui && npx vitest run src/logic/`
- [ ] Verify 4番30符 = 8000 (mangan, not 7700)
- [ ] Verify 暗杠中张 = 16符 (not 8)
- [ ] Verify pinfu detected via liangmian when kanzhang also possible
- [ ] Verify calculator UI still works with renamed types

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: ui/src/logic/riichi/__tests__/fu.test.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import { Meld } from '../types'
+import { tiles, defaultOptions } from './test-helpers'
+
+describe('Fu Calculation', () => {
+  it('chiitoitsu always 25fu', () => {
+    const hand = tiles('1133m2277p4488s6z')
+    const win = Tile.fromString('6z')
+    const result = calculateHand(hand, [], win, defaultOptions())
+    expect(result!.fu).toBe(25)
+  })
```
[the remark this answers is not in the document]
Comment (mmdtoycar) [reply ui/src/logic/riichi/__tests__/fu.test.ts:14]: Fixed in d299617. Added null checks and removed unused groups parameter.

File: ui/src/logic/riichi/__tests__/scoring.test.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      const result = calculateHand(hand, [], win, defaultOptions())
+      expect(result!.score.ron).toBe(32000)
+    })
+
+    it('yakuman dealer ron = 48000', () => {
+      const hand = tiles('19m19p19s1234567z')
+      const win = Tile.fromString('1m')
+      const result = calculateHand(hand, [], win, defaultOptions({ zifeng: 1 }))
+      expect(result!.score.ron).toBe(48000)
+    })
```
[the remark this answers is not in the document]
Comment (mmdtoycar) [reply ui/src/logic/riichi/__tests__/scoring.test.ts:97]: Fixed in d299617. Added null checks.
